File: scripts/Sum Closest/Sum Closest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
    // TC : O(n^2)
    const n = nums.length;
    nums.sort((x, y) => x - y);
    var result = nums[1]+nums[0]+nums[2]
    
    for(var i = 0 ; i< n-1 ; i++)
    {
        var low = i+1;
        var high = n-1;
        
        while(low < high){
            var sum = nums[i] + nums[low] + nums[high];
            
            if(target == sum)
                return sum;
            
            if(Math.abs(sum-target) < Math.abs(result - target)){
                result = sum;
            }
            
            if(sum > target)
                high -= 1;
            
            else
                low += 1;
            
        }
    }
    return result;
};

File: scripts/algorithms/Sum Closest/Sum Closest.java
               class Solution {
public int threeSumClosest(int[] nums, int target) {
    int n=nums.length;
    int count=0;
    int sumone=0;
     int value =Integer.MAX_VALUE;
    Arrays.sort(nums);
    for(int i=0; i<n; i++){
        for(int j=i+1; j<n; j++ ){
            for(int k=j+1; k<n; k++){
                count= nums[i]+ nums[j]+ nums[k];
			   if(count==target){
                    return count;
                }
                 int diff = target-count;
             if(Math.abs(diff)<value){
                value=Math.abs(diff);
                sumone=count;
            }
        }
    }
}
    return sumone;
	}
